File: src/mem/ruby/network/garnet/fixed-pipeline/RoutingTableEntry.cc
#include "mem/ruby/network/garnet/fixed-pipeline/RoutingTableEntry.hh"

RoutingTableEntry::RoutingTableEntry(const Params *p)
	: SimObject(p)
{
	m_cur_router = p->cur_router;
	m_dest_router = p->dest_router;
	m_invc = p->invc;
	m_next_router = p->next_router;
	m_outvc = p->outvc;
}

RoutingTableEntry *
RoutingTableEntryParams::create()
{
	return new RoutingTableEntry(this);
}

File: src/mem/ruby/network/garnet/fixed-pipeline/RoutingTableEntry.py
from m5.params import *
from m5.SimObject import SimObject

class RoutingTableEntry(SimObject):
    type = 'RoutingTableEntry'
    cxx_header = "mem/ruby/network/garnet/fixed-pipeline/RoutingTableEntry.hh"
    cur_router = Param.Int("Current router");
    dest_router = Param.Int("Destination router");
    invc = Param.Int("Input virtual channel");
    next_router = Param.Int("Next router");
    outvc = Param.Int("Output virtual channel");

    
